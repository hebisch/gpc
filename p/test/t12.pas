program records(output);
type
        cmplx = record
                rp: real;
                ip: real;
        end;
var
        x, y: cmplx;
function cadd(a,b: cmplx): cmplx;
        var result: cmplx;
        begin
                result.rp := a.rp + b.rp;
                result.ip := a.ip + b.ip;
                cadd := result;
        end;
procedure cprint(a: cmplx);
        begin
                writeln(a.rp, '+', a.ip, 'i');
        end;
begin
        x.rp := 1;
        x.ip := 2;
        y.rp := 3;
        y.ip := 4;
        cprint(cadd(x, y));
end.
