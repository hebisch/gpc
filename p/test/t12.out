 4.000000000000000e+00+ 6.000000000000000e+00i
